Parser/ContainerCallVisitor.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+
+namespace ChameleonSystem\UpgradeHelperBundle\Parser;
+
+
+use ChameleonSystem\UpgradeHelperBundle\DataModel\CallToContainer;
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\NodeVisitorAbstract;
+
+/**
+ * ContainerCallVisitor will visit every call to a container/service locator `get` method and produce
+ * a `CallToContainer` instance for every occurrence.
+ */
+final class ContainerCallVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var CallToContainer[]
+     */
+    private $calls;
+    /**
+     * @var string
+     */
+    private $fileName;
+    /**
+     * @var TypeResolveVisitor
+     */
+    private $typeResolver;
+
+    public function __construct(TypeResolveVisitor $typeResolver, string $fileName)
+    {
+        $this->calls = [];
+        $this->fileName = $fileName;
+        $this->typeResolver = $typeResolver;
+    }
+
+
+    public function leaveNode(Node $node)
+    {
+        if ($node instanceof Node\Expr\MethodCall) {
+            if (!property_exists($node->name, 'name')) {
+                return null;
+            }
+            if ($node->name->name !== 'get') {
+                return null;
+            }
+
+            // calling on attribute of `this`
```

Comment:
Is there a deeper reason why we only check calls on `$this`? That would potentially miss invocations such as this:

```
$container = $this->container;
$container->get('foo');
```

Reply:
Actually, that might not be as easy as I thought since we only keep track of the types of class properties, not single variables. I'll check out the code and see what php parser has to offer to potentially help here.

Reply:
:ok: Sadly, It's not as easy as I thought it'd be. Sad.